File: app/src/main/java/co/appdev/boilerplate/ui/splash/SplashActivity.java
package co.appdev.boilerplate.ui.splash;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import java.util.concurrent.TimeUnit;

import co.appdev.boilerplate.R;
import co.appdev.boilerplate.ui.main.MainActivity;
import io.reactivex.Maybe;

import static co.appdev.boilerplate.util.Constants.SPLASH_TIME_OUT;


public class SplashActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        Maybe.empty()
                .delay(SPLASH_TIME_OUT, TimeUnit.SECONDS)
                .doOnComplete(() -> {
                    startActivity(MainActivity.getStartIntent(SplashActivity.this,true));
                    finish();
                })
                .subscribe();
    }
}

File: app/src/main/java/co/appdev/lms/ui/SplashActivity.kt
package co.appdev.lms.ui

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import co.appdev.lms.R
import co.appdev.lms.util.Constants
import com.microsoft.appcenter.AppCenter
import com.microsoft.appcenter.analytics.Analytics
import com.microsoft.appcenter.crashes.Crashes
import io.reactivex.Maybe
import java.util.concurrent.TimeUnit


class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.splash_activity)
        AppCenter.start(
            getApplication(), Constants.AppCenter_Secret,
            Analytics::class.java, Crashes::class.java
        )
        val uiOptions = View.SYSTEM_UI_FLAG_HIDE_NAVIGATION or View.SYSTEM_UI_FLAG_FULLSCREEN
        window.decorView.systemUiVisibility = uiOptions
        Maybe.empty<Any>()
            .delay(Constants.SPLASH_TIME_OUT, TimeUnit.MILLISECONDS)
            .doOnComplete {
                startActivity(MainActivity.getStartIntent(this))
                finish()
            }.subscribe()
    }
}
